spec/rails_helper.rb

```diff
@@ -214,10 +212,20 @@ def seed_base_data_for_tests
     Capybara.server = :puma, { Silent: true }
   end
 
-  config.before(:each) do
-    # Defined shared @ global variables used throughout the test suite.
-    define_global_variables if RSpec.current_example.metadata[:seed_items] != false
+  config.before(:all) do
+    unless Thread.current[:skipped_last_seeding]
+      DatabaseCleaner.clean_with(:truncation)
```

Comment:
The logic here is a bit abstract unless someone has been involved with our discussions (and even I'm having a hard time keeping it straight). Can we make it really explicit?

*   If current and previous metadata are equal, we don't have to take any action at all
*   If current metadata says seed and previous metadata didn't, we have to seed the data
*   If previous metadata said seed and current one doesn't, we have to truncate

The double negative of "if not skip" is hard to follow for me. We have to do it this way ("skip" instead of "seed") to avoid changing all our specs at once, but let's try to make it as clear as possible.

Reply:
That's a totally fair point. I think its actually confusing enough I think I got the logic wrong here. 
This is our goal right?

![image](https://github.com/rubyforgood/human-essentials/assets/14540596/c0206162-c0c2-44ba-a326-e4fd335da91c)